cli/azd/internal/appdetect/docker.go

```diff
@@ -1,19 +1,75 @@
 package appdetect
 
 import (
+	"bufio"
+	"fmt"
 	"io/fs"
+	"log"
+	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
 )
 
 func detectDocker(path string, entries []fs.DirEntry) (*Docker, error) {
 	for _, entry := range entries {
 		if strings.ToLower(entry.Name()) == "dockerfile" {
+			dockerFilePath := filepath.Join(path, entry.Name())
+			ports, err := parsePortsInFile(dockerFilePath)
+			if err != nil {
+				return nil, fmt.Errorf("parsing ports in file %s: %w", dockerFilePath, err)
+			}
 			return &Docker{
-				Path: filepath.Join(path, entry.Name()),
+				Path:  dockerFilePath,
+				Ports: ports,
 			}, nil
 		}
 	}
 
 	return nil, nil
 }
+
+func parsePortsInFile(dockerFilePath string) ([]Port, error) {
```

Comment:
I think I liked the previous iteration, where we didn't separate out `parsePortsInFile` as a function.

The main reason why I'd prefer the old version is that in this new version, we would introduce code churn (additional background changes) when we say, for example, parse `RUN` from the Dockerfile.

If we move the logic back in `detectDocker`, I'd feel much better about it.

Reply:
Updated.